dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java

```diff
@@ -0,0 +1,103 @@
+import com.dotcms.mock.request.MockAttributeRequest;
+import com.dotcms.mock.request.MockHeaderRequest;
+import com.dotcms.mock.request.MockHttpRequest;
+import com.dotcms.mock.request.MockSessionRequest;
+import com.dotcms.mock.response.MockHttpResponse;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.beans.Permission;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.liferay.portal.model.User;
+import com.liferay.portal.util.WebKeys;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import org.glassfish.jersey.internal.util.Base64;
+import static org.junit.Assert.*;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class UserResourceIntegrationTest {
+
+    static HttpServletResponse response;
+    static HttpServletRequest request;
+    static UserResource resource;
+    static User user;
+    static Host host;
+    static User adminUser;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+        // Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+
+        resource = new UserResource();
+        adminUser = TestUserUtils.getAdminUser();
+        host = new SiteDataGen().nextPersisted();
+        user = TestUserUtils.getChrisPublisherUser(host);
+        response = new MockHttpResponse();
+
+        //Check if role has any layout, if is empty add one
+        if(APILocator.getLayoutAPI().loadLayoutsForUser(user).isEmpty()) {
+            APILocator.getRoleAPI()
+                    .addLayoutToRole(APILocator.getLayoutAPI().findAllLayouts().get(0),
+                            APILocator.getRoleAPI().getUserRole(user));
+        }
+        //Add permissions to the host
+        final Permission readPermissionsPermission = new Permission( host.getPermissionId(),
+                APILocator.getRoleAPI().getUserRole(user).getId(), PermissionAPI.PERMISSION_READ, true );
+        APILocator.getPermissionAPI().save(readPermissionsPermission,host,adminUser,false);
+
+    }
+
+    private static HttpServletRequest mockRequest() {
+        final MockHeaderRequest request = new MockHeaderRequest(
+                new MockSessionRequest(
+                        new MockAttributeRequest(new MockHttpRequest("localhost", "/").request())
+                                .request())
+                        .request());
+
+        request.setHeader("Authorization",
+                "Basic " + new String(Base64.encode("admin@dotcms.com:admin".getBytes())));
+
+        request.getSession().setAttribute(com.dotmarketing.util.WebKeys.CURRENT_HOST,host);
+        request.getSession().setAttribute(com.dotmarketing.util.WebKeys.CMS_SELECTED_HOST_ID,host);
+
+        return request;
+    }
+
+    private void loginAs() throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)